src/components/helpdesk/repo/problem.repo.integration.test.js

```diff
@@ -0,0 +1,78 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const { ObjectId } = require('mongoose').Types;
+const appPromise = require('../../../app');
+const testObjects = require('../../../../test/services/helpers/testObjects')(appPromise);
+const { problemRepo } = require('.');
+const { GeneralError, ValidationError } = require('../../../errors');
+
+chai.use(chaiAsPromised);
+const { expect } = chai;
+
+describe('problem repo', () => {
+	let app;
+	let server;
+
+	before(async () => {
+		app = await appPromise;
+		server = await app.listen(0);
+	});
+
+	afterEach(async () => {
+		await testObjects.cleanup();
+	});
+
+	after(async () => {
+		await server.close();
+	});
+
+	describe('deleteProblems', () => {
+		it('when problem is deleted, it should return deleted problem ids', async () => {
+			const user = await testObjects.createTestUser();
+			const userId = user._id;
+			await testObjects.createTestProblem({ userId });
+			const result = await problemRepo.deleteProblemsForUser(userId);
+			expect(result.success).to.be.equal(true);
+			expect(result.deletedDocuments).to.be.equal(1);
+		});
+
+		it('when problem is deleted, it should be gone from db', async () => {
+			const user = await testObjects.createTestUser();
+			const userId = user._id;
+			await testObjects.createTestProblem({ userId });
+
+			let userProblems = await problemRepo.getProblemsForUser(userId);
+			expect(userProblems.length).to.be.equal(1);
```

Comment:
adding id check and create another test problem of another user which should exist after deletion